spec/rspec/core/world_spec.rb

```diff
@@ -136,6 +136,19 @@ module RSpec::Core
       context "when --only-failures is passed" do
         before { configuration.force(:only_failures => true) }
 
+        context "and all examples are filtered out" do
+          before do
+            configuration.filter_run_including :foo => 'bar'
+          end
+
+          it 'will ignore run_all_when_everything_filtered' do
+            allow(configuration).to receive(:run_all_when_everything_filtered?) { true }
```

Comment:
I'd prefer to see this do `configuration.run_all_when_everything_filtered = true`.  I don't think stubbing it really provides a benefit here.